ci/scripts/detect-docker-updates.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+latest_version=$(curl -I -s https://github.com/docker/docker-ce/releases/latest | grep "location:" | awk '{n=split($0, parts, "/"); print substr(parts[n],2);}' | awk '{$1=$1;print}' | tr -d '\r' | tr -d '\n' )
+
+if [[ $latest_version =~ (beta|rc) ]]; then
+	echo "Skip pre-release versions"
+	exit 0;
+fi
+
+latest="https://download.docker.com/linux/static/stable/x86_64/docker-$latest_version.tgz"
+current=$( git-repo/ci/images/get-docker-url.sh )
+
+if [[ $current = $latest ]]; then
+	echo "Already up-to-date"
+	exit 0;
+fi
+
+
+issue_title="Upgrade to Docker $latest_version in CI"
+
+existing_tasks=$( curl -s https://api.github.com/repos/${GITHUB_ORGANIZATION}/${GITHUB_REPO}/pulls\?labels\=type:%20task\&state\=open\&creator\=spring-buildmaster )
+existing_upgrade_issues=$( echo "$existing_tasks" | jq -c --arg TITLE "$issue_title" '.[] | select(.title==$TITLE)' )
+
+if [[ ${existing_upgrade_issues} = "" ]]; then
+	# Create changes in dedicated branch
+	branch="ci-docker-$latest_version"
+	git config user.name "Spring Buildmaster" > /dev/null
+	git config user.email "buildmaster@springframework.org" > /dev/null
+	git checkout -b "$branch" origin/master > /dev/null
```

Comment:
I'm not sure if this will work in the Concourse setup as I don't know if we have a real repository there.

Reply:
It will work. `stage.sh` does [something similar](https://github.com/spring-projects/spring-boot/blob/b66ca0016a8e08f0b86ea93d54112ce9d0fa50a5/ci/scripts/stage.sh#L35).